### DISCUSSION TITLE
Leap second smearing compatibility?

### DISCUSSION THREAD

**Discussion by mcpherrinm** [pendulum-project/ntpd-rs#1411] [Q&A] [answered]
It's increasingly popular to run time servers that smear leap seconds, such as on AWS and Google's public time servers.

What would happen if ntpd-rs is pointed at an NTP server which smears leap seconds?

What about a mix of smearing and non-smearing servers?

**Comment by davidv1992** [comment, marked as answer]
If ntpd-rs is pointed at time servers that do leap smearing, then it will "just" follow those time servers during the smear. Depending on the type of smearing used, some increased clock instability might occur around the start and end of the smear, but this should not be problematic for most applications (I would expect an additional error of 10-100 microseconds).

When pointing to a mix of smearing and non-smearing servers, or smearing servers that use different types of smearing, we will act like any NTP implementation. It will detect that the servers are in disagreement, and try to find a majority that agree. If it can find that, it will follow that majority. If not, it will likely stop synchronizing the local clock for the duration of the smear. Given that you started out in a synchronized situation, this will likely result in an error somewhere around the 0.5 second mark just before synchronization restarts, assuming 24 hour smear intervals.
